build-customization.rst

```diff
@@ -0,0 +1,72 @@
+.. post:: Nov 10, 2022
+   :tags: announcement, feature, builds
+   :author: Manuel
+   :location: BCN
+
+.. meta::
+   :description lang=en:
+      We released user-defined commands (``build.commands`` config key)
+      which allow users to overwrite the build process completely.
+
+
+Overwrite the build process completely
+======================================
+
+We are happy to announce a new *beta* feature that allow users to **overwrite the build process completely**.
+In the past, `we talked about executing custom commands <https://blog.readthedocs.com/user-defined-build-jobs/>`_ *in-between* the Read the Docs build process.
+However, that approach may not be enough for some projects with strong custom build processes
+and they were not able to use our platform because of that.
+Until now! We have good news for them!
+
+The new key ``build.commands`` allows projects to only execute *exactly* the commands they want.
+No more. No less.
+This means that Read the Docs *won't execute any of the default* commands behind the scenes.
+You have 100% control over the building process.
+
+
+How it works?
+-------------
+
+To use the new *beta* feature,
+you can specify all the commands the build process requires under the ``build.commands`` `config key <https://docs.readthedocs.io/en/stable/config-file/v2.html#build-commands>`_.
+Finally, move the resulting documentation under ``_readthedocs/html`` directory.
+Read the Docs will upload and host all the content of that folder ✨
+
+Here is simple example that builds a small project's documentation using `Docsify <https://docsify.js.org/>`_:
+
+.. code:: yaml
+
+   version: 2
+   build:
+     os: "ubuntu-22.04"
+     tools:
+       python: "3.11"
+     commands:
+       # Create the output directory
+       - mkdir --parents _readthedocs/html/
+       # Copy the documents (including the ``index.html``)
+       # into the output directory
+       - cp --recursive docs/* _readthedocs/html/
```

Comment:
Would you agree that ideally `_readthedocs/html/` is read from some environment variable, like a `$READTHEDOCS_HTTP_ROOT` or similar? Just wanted to know if it would be relevant to open an issue, I don't see anything listed here: https://docs.readthedocs.io/en/latest/environment-variables.html

Reply:
I don't have strong opinions about this. This is the default contract that we built some time ago when working on this.

What would be the benefit about defining an environment variable? We could avoid copying a bunch of files; but instead of `cp` we could just create a symlink as well, if that's the problem we want to avoid. I'm not sure if an environment variable will be a benefit for users or will be trickier instead.